plugins/Core-Extensions/ProductLine/edu.toronto.cs.se.mmint.productline.design/src/edu/toronto/cs/se/mmint/productline/design/ToProductLine.java

```diff
@@ -250,62 +243,148 @@ private ContainerMapping createProductLineContainerMapping(ContainerMapping orig
 	    return containerMapping;
 	}
   
-  private NodeCreationDescription createProductLineNodeCreationTool(NodeCreationDescription originalCreateTool, Map<String, Object> map) {
+  private NodeCreationDescription createProductLineNodeCreationTool(NodeCreationDescription originalCreateTool, Map<String, DiagramElementMapping> map) {
 	    var createTool = this.out.dToolFactory.createNodeCreationDescription();
 	    createTool.setName("PL" + originalCreateTool.getName());
 	    createTool.setLabel("PL" + originalCreateTool.getLabel());
-	    String newString = originalCreateTool.getLabel().replace(" ", "");
-	    /*
-	     * QUESTION 4
-	     */
+	    
 	    for (var originalMapping : originalCreateTool.getNodeMappings()) {
-	    	/*
-	    	 * QUESTION 5
-	    	 */
 	    	NodeMapping newMapping = (NodeMapping) map.get(originalMapping.getName());
 	    	createTool.getNodeMappings().add(newMapping);
 	    }
-	    
-	    createTool.setIconPath("/edu.toronto.cs.se.modelepedia.statemachine.edit/icons/full/obj16/"+newString+".gif");
-//	    originalCreateTool.getInitialOperation().getFirstModelOperations();
 	    /*
-	     * QUESTION 7
+	     * QUESTION 1: How do I add these to the actual creation tool? Are these the right elements in the first
+	     * place?
+	     */
+	    var container = this.out.vToolFactory.createNameVariable();
+	    // or do we use createSelectContainerVariable()? or createElementVariable()?
+	    var containerView = this.out.vToolFactory.createContainerViewVariable();
+	     
+        container.setName("container");
+        containerView.setName("containerView");
+//        createTool.getVariable().;
+        
+        /*
+         * QUESTION 2: We didn't discuss this in the meeting but I'm assuming the initial operation does need
+         * to be created before we can get it?
```

Comment:
We probably missed this, not sure without a computer.
The fact that you have to explicitly cast it though is not a good sign, it should just be assignable as-is.